File: Elearn.Data/Repository/CommentRepository.cs
using Elearn.Application.RepositoryInterfaces;
using Elearn.Data.Data;
using Shared.Models;

namespace Elearn.Data.Repository;

public class CommentRepository : ICommentRepository
{
    private DataContext _dataContext;

    public CommentRepository(DataContext dataContext)
    {
        _dataContext = dataContext;
    }
    public async Task<Comment> CreateAsync(Comment comment)
    {
        _dataContext.Add(comment);
        await _dataContext.SaveChangesAsync();
        return comment;
    }
}

File: Elearn-DataTier/src/main/java/via/dk/elearn/repository/CommentRepository.java
package via.dk.elearn.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import via.dk.elearn.models.Comment;
import via.dk.elearn.models.Lecture;

import java.util.List;

@Repository
public interface CommentRepository extends JpaRepository<Comment,Long> {

    List<Comment> findAllByLecture(Lecture lecture);
}
